cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/ConsumerConfiguration.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib.partitioned;
+
+import co.cask.cdap.api.Predicate;
+import co.cask.cdap.api.dataset.lib.PartitionDetail;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Configuration parameters to be used by a {@link PartitionConsumer}.
+ */
+public class ConsumerConfiguration {
+  /**
+   * Instance of ConsumerConfiguration with default parameters
+   */
+  public static final ConsumerConfiguration DEFAULT = builder().build();
+
+  private final Predicate<PartitionDetail> partitionPredicate;
```

Comment:
using a predicate that returns a boolean only allows you to accept or reject a partition and stop. In the future we may want to distinguish more than just 2 cases, for example, we may want to indicate that the partition should be skipped but we want to keep going. For that purpose it would be more extensible if the predicate can return an enum. Today, that enum might only have ACCEPT and STOP, but in the future we might add SKIP to that.